src/binary-reader-opcnt.cc

```diff
@@ -35,49 +35,37 @@ struct Context {
 
 }  // namespace
 
-static Result add_int_counter_value(IntCounterVector* vec,
-                                        intmax_t value) {
-  for (size_t i = 0; i < vec->size; ++i) {
-    if (vec->data[i].value == value) {
-      ++vec->data[i].count;
+static Result add_int_counter_value(IntCounterVector* vec, intmax_t value) {
+  for (size_t i = 0; i < vec->size(); ++i) {
+    if ((*vec)[i].value == value) {
+      ++(*vec)[i].count;
       return Result::Ok;
     }
   }
-  IntCounter counter;
-  counter.value = value;
-  counter.count = 1;
-  append_int_counter_value(vec, &counter);
+  vec->emplace_back(value, 1);
   return Result::Ok;
 }
 
 static Result add_int_pair_counter_value(IntPairCounterVector* vec,
-                                             intmax_t first,
-                                             intmax_t second) {
-  for (size_t i = 0; i < vec->size; ++i) {
-    if (vec->data[i].first == first && vec->data[i].second == second) {
-      ++vec->data[i].count;
+                                         intmax_t first,
+                                         intmax_t second) {
+  for (size_t i = 0; i < vec->size(); ++i) {
```

Comment:
Couldn't you use a simpler for loop here, since only one element is being accessed at a time.

for (IntPairCounter& counter : vec) { ... }